scripts/importDpsConstants.ts

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env npx tsx
+import { spawnSync } from 'node:child_process'
+import fs from 'node:fs'
+import path from 'node:path'
+
+import type { IncidentReportingApi, Constant, TypeConstant } from '../server/data/incidentReportingApi'
+
+type ConstantsMethod = keyof IncidentReportingApi['constants']
+interface Template {
+  method: ConstantsMethod
+  enumName: string
+  documentation?: string
+}
+const templates: Template[] = [
+  { method: 'types', enumName: 'Type' },
+  { method: 'statuses', enumName: 'Status' },
+  { method: 'informationSources', enumName: 'InformationSource' },
+  { method: 'staffInvolvementRoles', enumName: 'StaffInvolvementRole' },
+  { method: 'prisonerInvolvementRoles', enumName: 'PrisonerInvolvementRole' },
+  { method: 'prisonerInvolvementOutcomes', enumName: 'PrisonerInvolvementOutcome' },
+  { method: 'correctionRequestReasons', enumName: 'CorrectionRequestReason' },
+  {
+    method: 'errorCodes',
+    enumName: 'ErrorCode',
+    documentation: 'Unique codes to discriminate errors returned from the incident reporting api',
+  },
+]
+
+function printHelp(): never {
+  const help = `
+Imports constants from incident reporting api downloaded as JSON files.
+Usage:
+  ./scripts/importDpsConstants.ts [type] [file path]
+
+Where [type] is one of ${templates.map(template => template.method).join(', ')}
+`.trim()
+  process.stderr.write(`${help}\n`)
+  process.exit(1)
+}
+
+const [, , type, filePath] = process.argv
+if (!type || !filePath) {
+  printHelp()
+}
+const template = templates.find(t => t.method === type)
+if (!template) {
+  printHelp()
+}
+const { method, enumName, documentation } = template
+
+const constants: Constant[] = JSON.parse(fs.readFileSync(filePath, { encoding: 'utf8' }))
+
+const outputPath = path.resolve(__dirname, `../server/reportConfiguration/dpsConstants/${method}.ts`)
+const outputFile = fs.openSync(outputPath, 'w')
+
+fs.writeSync(outputFile, `// Generated with ./scripts/importDpsConstants.ts ${new Date().toISOString()}\n\n`)
+if (method === 'errorCodes') {
+  // error codes are numbers so need special treatment
+
+  if (documentation) {
+    fs.writeSync(outputFile, `/** ${documentation} */\n`)
+  }
+  fs.writeSync(outputFile, '// eslint-disable-next-line import/prefer-default-export\n')
+  fs.writeSync(outputFile, `export enum ${enumName} {\n`)
+  constants.forEach(constant => {
+    fs.writeSync(outputFile, `${constant.description} = ${constant.code},\n`)
+  })
+  fs.writeSync(outputFile, '}\n')
+} else if (method === 'types') {
+  // types are string constants but have extra info
+
+  fs.writeSync(outputFile, `export const ${method} = [\n`)
+  constants.forEach((constant: TypeConstant) => {
+    fs.writeSync(outputFile, `/** ${constant.description} */\n`)
```

Comment:
wouldn't be nicer to have the description as a machine-readable field? (I don't care too much for these doc comments in this context as the values themselves are pretty self-explanatory between the code and the description)

Reply:
and it may be used somehow by the forms/etc...for example when showing a select the `option` value would be the code but presumably the inner text for human would be this description?

Reply:
> these are here just for the sake of tooltips in the code editor :)
oh I see, in that case leave the doc comments :)

I think it's worth adding the description field, it's information the API has and it makes sense to expose it in the UI given we're already exporting the values anyway.